### DISCUSSION TITLE
Docker Swarm with Infisical agent

### DISCUSSION THREAD

**Discussion by SystematicCZ** [Infisical/infisical#2174] [Q&A] [answered]
Hi,

I have a question about the Infisical Agent running on Docker Swarm. In the docs, there is a volume created for storing secrets, but if I have multiple nodes, volumes are not shared. Should the Infisical Agent be a global service running on each node, or what is the general guideline to solve this?

**Comment by maidul98** [comment, marked as answer]
The idea is you want Infisical Agent to drop the secrets for your application to consume. Since docker swarm doesn't have global storage, running on each node would make sense as applications can pick up secrets from the designated file.
